### DISCUSSION TITLE
Command-line samples and specifying a different package name (with a dash)

### DISCUSSION THREAD

**Discussion by dblock** [python-poetry/poetry#8108] [Q&A]
This is a bit similar to https://github.com/orgs/python-poetry/discussions/4092, but with a twist.

I have a basic poetry setup for a project called `package` in https://github.com/dblock/poetry-sample. Both my tests and sample import `from package.something import Something`. I can run tests with `poetry run pytest` and a sample with `poetry run samples/package/hello.py`. 

I'd actually like the package to be called `my-package`. If I rename it in `pyproject.toml` as follows.

```toml
[tool.poetry]
name = "my-package"
```

Then the tests still run, but the sample no longer finds `package`.

```
$ poetry run samples/package/hello.py 
/Users/dblock/source/dblock/poetry-sample/samples/package
/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python311.zip
/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11
/opt/homebrew/Cellar/python@3.11/3.11.3/Frameworks/Python.framework/Versions/3.11/lib/python3.11/lib-dynload
/Users/dblock/Library/Caches/pypoetry/virtualenvs/my-package--Lq8I24Z-py3.11/lib/python3.11/site-packages
Traceback (most recent call last):
  File "/Users/dblock/source/dblock/poetry-sample/samples/package/hello.py", line 8, in <module>
    from package.something import Something
ModuleNotFoundError: No module named 'package'
```

I believe this should be possible. I am aiming to publish a package called `my-package` on pypi, but I want users to import `package`. 

How do I do that, or am I missing something?

**Comment by dimbleby** [comment]
https://python-poetry.org/docs/pyproject/#packages

**Reply by dblock** [reply]
I could not make it work. If you want to contribute a change that does work to https://github.com/dblock/poetry-sample that'd be awesome.
